### PR TITLE
Add number platform to Tesla Fleet

### PR DESCRIPTION
**Pull request by Bre77**
## Proposed change

Add number platform to Tesla Fleet

## Type of change

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [ ] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logos and icons in [Brands repository](https://github.com/home-assistant/brands).
- [x] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information

- Link to parent pull request in the codebase: https://github.com/home-assistant/core/pull/125985
- Link to parent pull request in the Brands repository: 
- This PR fixes or closes issue: fixes #

## Checklist

- [x] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [x] The documentation follows the Home Assistant documentation [standards].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

## Summary by CodeRabbit

- **New Features**
	- Introduced new entity types for the Tesla Fleet integration, enhancing the ability to monitor vehicle charging parameters.
	- Added detailed insights with new entities: Charge current, Charge limit, Backup reserve, and Off grid reserve, improving energy management capabilities.
	- Added a new "Number" category and platform for more granular control and monitoring of vehicle performance.

### CODE REVIEW

File: source/_integrations/tesla_fleet.markdown [hunk 1]
```diff
@@ -143,6 +147,8 @@ These are the entities available in the Tesla Fleet integration. Not all entitie
 |Binary sensor|Backup capable|Yes|
 |Binary sensor|Grid services active|Yes|
 |Binary sensor|Grid services enabled|Yes|
+|Number|Backup reserve|Yes|
```
Comment (c0ffeeca7) [inline source/_integrations/tesla_fleet.markdown:160]: what is meant by reserve?

Comment (Bre77) [reply source/_integrations/tesla_fleet.markdown:160]: https://www.tesla.com/en_au/support/energy/powerwall/mobile-app/backup-reserve

Comment (c0ffeeca7) [reply source/_integrations/tesla_fleet.markdown:160]: Thank you

**Review by c0ffeeca7 — DISMISSED** [review]
Thank you, @Bre77 :+1:

**Review by joostlek — DISMISSED** [review]
There's a merge conflict

**Review by c0ffeeca7 — APPROVED** [review]

**Pull request merged**